File: src/main/java/itmo/is/project/dto/resource/ResourceAmountDto.java
package itmo.is.project.dto.resource;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;

public record ResourceAmountDto(

        @JsonProperty(value = "resource", required = true)
        ResourceDto resource,

        @Schema(example = "500")
        @JsonProperty(value = "amount", required = true)
        Integer amount
) {
}

File: src/main/kotlin/itmo/isproject/shared/resource/dto/ResourceAmountDto.kt
package itmo.isproject.shared.resource.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

data class ResourceAmountDto(

    @JsonProperty(value = "resource", required = true)
    val resource: ResourceDto,

    @Schema(example = "500")
    @JsonProperty(value = "amount", required = true)
    val amount: Int?
)
